front/lib/swr/conversations.ts

```diff
@@ -567,3 +569,110 @@ export function useConversationMarkAsRead({
     markAsRead,
   };
 }
+
+type ConversationAction = "join" | "leave" | "delete";
+
+export const useConversationMenu = ({
+  ownerId,
+  conversationId,
+  userId,
+}: {
+  ownerId: string;
+  conversationId: string | null;
+  userId: string | null;
+}): {
+  joinConversation: () => Promise<boolean>;
+  action: ConversationAction;
+  users: UserParticipantType[];
+  agents: AgentParticipantType[];
+} => {
+  const sendNotification = useSendNotification();
+
+  const { mutateConversations } = useConversations({
+    workspaceId: ownerId,
+  });
+  const { mutateConversationParticipants, conversationParticipants } =
+    useConversationParticipants({
+      conversationId,
+      workspaceId: ownerId,
+      options: { disabled: conversationId === null },
+    });
+
+  const isUserParticipating =
+    userId !== null &&
+    conversationParticipants?.users.find(
+      (participant) => participant.sId === userId
+    );
+
+  const isLastParticipant =
+    isUserParticipating && conversationParticipants?.users.length === 1;
+
+  const joinConversation = useCallback(async (): Promise<boolean> => {
+    if (!conversationId) {
+      return false;
+    }
+    try {
+      const response = await fetch(
+        `/api/w/${ownerId}/assistant/conversations/${conversationId}/participants`,
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        }
+      );
+
+      if (!response.ok) {
+        if (response.status === 400) {
+          const error = await response.json();
+          if (error.error?.message?.includes("already a participant")) {
```

Comment:
we can maybe try to add a new api_error.type instead of trying to match the code + message
(see API_ERROR_TYPES array)